stmt/stmts/select.go

```diff
@@ -149,7 +149,7 @@ func (s *SelectStmt) Plan(ctx context.Context) (plan.Plan, error) {
 		}
 	}
 	lock := s.Lock
-	if variable.ShouldAutocommit(ctx) {
+	if lock != coldef.SelectLockNone && ctx.(autocommit.Checker).ShouldAutocommit(ctx) {
```

Comment:
The only context we have is session. So we do not need to check it.
I use interface type asserting to solve dependence issues.